### PR TITLE
feat: rejection automation

### LINKED ISSUE #41: feat: rejection automation
We want to automate large portions of the rejection process, to make the review process simpler and more streamlined. 
We already have support for instant rejection of projects that violate our basic requirements.

After conducting an RFC, the following features were requested:
- [x] Reviewers (all members of the private feedback thread) should be pinged in feedback threads
- [x] Creation of feedback threads will be automated
- [x] Feedback messages should be posted by a human reviewer, not the bot
- [x] The bot should facilitate the drafting of review messages and allow edits by anyone

### PR DESCRIPTION
**Pull request by nullishamy**
Closes #41

### CODE REVIEW

**Comment by nullishamy** [comment]
Still to do: drafting support within the bot.

File: prisma/schema.prisma [hunk 1]
```diff
@@ -44,13 +44,25 @@ model Submission {
...
 
   // It might not exist, without circular dependencies, we cannot model
   // a schema where this is required.
   details SubmissionDetails?
 }
 
+model Draft {
+  content String
+
+  id Int @default(autoincrement())
```
Comment (BobobUnicorn) [inline prisma/schema.prisma:58]: make the id generation consistent (cuid or uuid seem appropriate)

File: prisma/schema.prisma [hunk 2]
```diff
@@ -44,13 +44,25 @@ model Submission {
...
+
+  submissionId String
+  submission   Submission @relation(fields: [submissionId], references: [id], onDelete: Cascade)
+
+  @@id([id, submissionId])
```
Comment (BobobUnicorn) [inline prisma/schema.prisma:63]: just id

File: src/commands/draft.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    builder.addSubcommand(cmd =>
+      cmd.setName('show')
+        .setDescription('Show the latest draft for a project')
+        .addStringOption(submissionNameStringOption)
+    )
+  },
+  shouldPublishGlobally: true,
+  onAutocomplete: submissionNameAutocompleteProvider,
+  onModalSubmit: async (client, interaction) => {
+    const [,submissionId] = interaction.customId.split('|')
```
Comment (BobobUnicorn) [inline src/commands/draft.ts:29]: factor out the customId stuff

File: prisma/schema.prisma [hunk 4]
```diff
@@ -44,13 +44,25 @@ model Submission {
...
   links           SubmissionLink[]
+  drafts          Draft[]
 
 
   // It might not exist, without circular dependencies, we cannot model
   // a schema where this is required.
   details SubmissionDetails?
 }
 
+model Draft {
```
Comment (BobobUnicorn) [inline prisma/schema.prisma:55]: also add timestamp for this draft

File: src/managers/submissions.ts [hunk 5]
```diff
@@ -121,7 +121,12 @@ export class SubmissionsManager extends Manager {
         links: true,
         votes: true,
         details: true,
-        relatedMessages: true
+        relatedMessages: true,
+        drafts: {
+          orderBy: {
+            id: 'desc'
```
Comment (BobobUnicorn) [inline src/managers/submissions.ts:127]: order by timestamp

File: src/utils/actionVotes.ts [hunk 6]
```diff
@@ -282,6 +299,84 @@ export async function submissionCleanup (submission: Submission, client: Project
   }
 }
 
+async function handleRejectionFeedback (submission: Submission, actioningVote: Vote, client: ProjectsClient): Promise<Result<void, Error>> {
+  const finalDraft = submission.drafts.currentDraft()
+
+  // Callers should check this
+  assert(finalDraft !== undefined, 'no draft was set')
+
+  let rejectionMessage = `<@${actioningVote.voter.id}>, please send the message belown to this thread:`
```
Comment (BobobUnicorn) [inline src/utils/actionVotes.ts:308]: use template string pls thx <3

File: src/utils/actionVotes.ts [hunk 7]
```diff
@@ -282,6 +299,84 @@ export async function submissionCleanup (submission: Submission, client: Project
...
+  }
+
+  rejectionMessage += '\n\n'
+
+  rejectionMessage += 'CC: '
+  rejectionMessage += [...privateReviewThread.members.cache.values()]
+    // Don't include the actioning voter, they're already here. Also don't include the bots
+    .filter(m => m.id !== actioningVote.voter.id && !m.user?.bot)
+    .map(m => `<@${m.id}>`)
+    .join(' ')
```
Comment (BobobUnicorn) [inline src/utils/actionVotes.ts:338]: factor this bit out?

File: src/utils/actionVotes.ts [hunk 8]
```diff
@@ -282,6 +299,84 @@ export async function submissionCleanup (submission: Submission, client: Project
...
+
+  rejectionMessage += '```'
+
+  // FIXME: change to private thread
```
Comment (BobobUnicorn) [inline src/utils/actionVotes.ts:342]: fix

File: src/utils/actionVotes.ts [hunk 9]
```diff
@@ -282,6 +299,84 @@ export async function submissionCleanup (submission: Submission, client: Project
...
+  const publicReviewThreadRes = await Result.wrapAsync<ThreadChannel, Error>(async () => await publicFeedback.threads.create({
+    name: submission.name,
+    type: 'GUILD_PUBLIC_THREAD'
```
Comment (BobobUnicorn) [inline src/utils/actionVotes.ts:345]: idea: set it as a config env

**Review by BobobUnicorn — CHANGES_REQUESTED** [review]

File: src/utils/actionVotes.ts [hunk 10]
```diff
@@ -328,21 +317,38 @@ async function handleRejectionFeedback (submission: Submission, actioningVote: V
...
+    const members = values
+      .filter(m => m.id !== actioningVote.voter.id && !m.user?.bot)
+      .map(m => `<@${m.id}>`)
+      .join(' ')
+
+    return `CC: ${members}`
+  }
 
-  rejectionMessage += '```'
+  // Callers should check this
```
Comment (BobobUnicorn) [inline src/utils/actionVotes.ts:336]:
```suggestion
  // Callers should already have checked this
```

**Review by BobobUnicorn — APPROVED** [review]

**Pull request merged**